### DISCUSSION TITLE
FirebaseError: Firebase: No Firebase App '[DEFAULT]' has been created

### DISCUSSION THREAD

**Discussion by borjalo** [gladly-team/next-firebase-auth#526] [Q&A] [answered]
I'm getting this error in my `login.tsx` file:

<img width="956" alt="image" src="https://user-images.githubusercontent.com/22989690/184111129-7a2b196e-ed9e-479e-a294-524155760b7c.png">

This is my `_app.tsx` file:

![image](https://user-images.githubusercontent.com/22989690/184110733-67c93e63-2c35-4cad-8c9c-f540fd73a4b4.png)

and this is my `login.tsx` file:

![image](https://user-images.githubusercontent.com/22989690/184110858-965b8d9e-b7d7-4370-a655-4d332caf090b.png)

Any idea of what is happening?

**Comment by borjalo** [comment]
I just needed to add `initializeApp({...})` from the Firebase JS SDK in the `_app.tsx` file and it worked. I don't know why because I thought that `initAuth()` was already initializing the Firebase app. Weird...

**Comment by borjalo** [comment]
I think that something is not working properly. When I try to access the auth object from firebase/auth in my login.tsx it seems to not be initialized yet so that's why it's crashing. How can I wait until the initAuth() in executed before calling the getAuth() from Firebase SDK?

**Reply by kmjennison** [reply]
Could you try setting up the demo app to see if it works for you? [v0.x example app ](https://github.com/gladly-team/next-firebase-auth/tree/main/example) / [v1.x example app ](https://github.com/gladly-team/next-firebase-auth/tree/v1.x/example)

The init method is synchronous and should work when used in _app.js. Are you sure your `utils/auth` module is behaving correctly?

If the above don't help, please share a minimal app with a reproduction of the issue.

**Reply by borjalo** [reply]
I think the demo app is not using custom login UI. Anyway, I tried it and the first page you access in the demo app is not getting the auth object, that's why it's working.

My `utils/auth` is initializing Firebase correctly because if I comment the `const auth = getAuth()` line and reload the page, it works fine.

I will try to share a minimal app reproducing the issue.

**Reply by borjalo** [reply]
@kmjennison I've created this blank repository with next-firebase-auth and, as I am experimenting in my project, when you try to call `const auth = getAuth()` in the page, the app crashes as it says that there's no app initialized. Take a look and keep me posted. Thanks!

https://github.com/borjalo/next-firebase-auth-test

**Reply by TobiasKrok** [reply]
Did you solve this issue?

**Reply by borjalo** [reply]
@TobiasKrok not yet. I'm waiting for a response. I think it's a bug in the package.

**Reply by kmjennison** [reply]
@borjalo I see—thanks for the additional info. You're currently trying to use the Firebase JS SDK during server-side rendering, but it's meant for use on the client side. You'll need to use `useEffect`:

```jsx
import React, { useEffect } from 'react'
import { getAuth } from 'firebase/auth'

const ExamplePage = () => {
  useEffect(() => {
    const auth = getAuth()
    // ...
  })
  // ...
}
```

[Here's the example](https://github.com/gladly-team/next-firebase-auth/tree/v1.x#using-the-firebase-apps) we have in docs.

Let me know if this helps.

**Reply by borjalo** [reply]
@kmjennison thanks for your response. Makes sense. I don't know why I didn't figure that out 😅.

**Comment by TobiasKrok** [comment]
I solved this by using React Contexts. I have not built my app completely so I'm not sure how it will work later with queries and such, but using this code you can access what you need from firebase in your react components

FirebaseContext.tsx
```javascript
import { FirebaseApp, initializeApp } from "firebase/app";
import { Auth, getAuth } from "firebase/auth";
import { Firestore, getFirestore } from "firebase/firestore";
import {
  createContext,
  PropsWithChildren,
  useContext,
  useEffect,
  useState,
} from "react";


type FirebaseState = {
  firebase: FirebaseApp;
  firestore: Firestore;
  auth: Auth;
};

const FirebaseContext = createContext<FirebaseState>({} as FirebaseState);

export const FirebaseProvider: React.FC<PropsWithChildren> = ({ children }) => {
  const [firebase, setFirebase] = useState<FirebaseApp>();
  const [firestore, setFirestore] = useState<Firestore>();
  const [auth, setAuth] = useState<Auth>();

  useEffect(() => {
    setFirebase(initializeApp(FIREBASE_CONFIG));
    setFirestore(getFirestore());
    setAuth(getAuth());
  }, []);

  return (
    <FirebaseContext.Provider
      value={{ firebase, firestore, auth } as FirebaseState}
    >
      {children}
    </FirebaseContext.Provider>
  );
};

export const useFirestore = () => {
  const { firestore } = useContext(FirebaseContext);
  return firestore;
};

export const useFirebaseAuth = () => {
  const { auth } = useContext(FirebaseContext);
  return auth;
};
export const useFirebase = () => {
  const { firebase } = useContext(FirebaseContext);
  return firebase;
};

```
_app.tsx
```javascript
export default App(props: Props) {
    const { Component, pageProps } = props;

    return (
    <FirebaseProvider>
        <Component {...pageProps}/>
    </FirebaseProvider>
    )
```
You can then use whichever hooks you need to do your authentication or firestore querying
login,tsx
```javascript
const Login = () => {
  const auth = useFirebaseAuth();
  //.... auth logic 
  return (
    <PaddedLayout>
      <Typography variant="h3">Login</Typography>
    </PaddedLayout>
  );
};
export default Login;
```

You can of course use Redux or whatever you want instead of contexts

**Reply by borjalo** [reply]
You're not using next-firebase-auth in your solution. It will work, but without SSR.

**Comment by kmjennison** [comment, marked as answer]
You're currently using the Firebase JS SDK during server-side rendering, but it's meant for use on the client side. You'll need to use `useEffect`:

```jsx
import React, { useEffect } from 'react'
import { getAuth } from 'firebase/auth'

const ExamplePage = () => {
  useEffect(() => {
    const auth = getAuth()
    // ...
  })
  // ...
}
```

[Here's the example](https://github.com/gladly-team/next-firebase-auth/tree/v1.x#using-the-firebase-apps) we have in docs.

_(Taken from [this comment thread](https://github.com/gladly-team/next-firebase-auth/discussions/526#discussioncomment-3449787) to make it easier for other people to find the answer.)_
